library/std/src/sys/pal/unix/process/process_unix.rs

```diff
@@ -640,6 +681,46 @@ impl Command {
             #[cfg(target_os = "nto")]
             let spawn_fn = retrying_libc_posix_spawnp;
 
+            if self.get_create_pidfd() {
+                let mut pidfd: libc::c_int = -1;
+                let spawn_res = pidfd_spawnp.get().unwrap()(
+                    &mut pidfd,
+                    self.get_program_cstr().as_ptr(),
+                    file_actions.0.as_ptr(),
+                    attrs.0.as_ptr(),
+                    self.get_argv().as_ptr() as *const _,
+                    envp as *const _,
+                );
+
+                let spawn_res = cvt_nz(spawn_res);
+                if let Err(ref e) = spawn_res
+                    && e.raw_os_error() == Some(libc::ENOSYS)
+                {
+                    PIDFD_SPAWN_SUPPORTED.store(NO, Ordering::Relaxed);
+                    return Ok(None);
+                }
+                spawn_res?;
+
+                let pid = match cvt(pidfd_getpid.get().unwrap()(pidfd)) {
+                    Ok(pid) => pid,
+                    Err(e) => {
+                        // The child has been spawned and we are holding its pidfd.
+                        // But we cannot obtain its pid even though pidfd_getpid support was verified earlier.
+                        // This might happen if libc can't open procfs because the file descriptor limit has been reached.
+                        libc::close(pidfd);
+                        return Err(Error::new(
+                            e.kind(),
+                            "pidfd_spawnp succeeded but the child's PID could not be obtained",
+                        ));
```

Comment:
In their unfathomable wisdom the glibc developers have not designed `pidfd_spawnp` to return the pid and pidfd at the same time even though `clone3` does support it, which means we can get into this awkward halfway state.

I'm not sure if this is the best solution for the edge-case. An alternative is to change it to  `Process { pid: Option<NonZero<pid_t>> }`  and delay the error reporting by only panicking if someone tries to call `Child.id()` if the when a pid couldn't be obtained.

Reply:
lmao.

Reply:
I think for right now this is best because I think assuming calling `.id()` won't panic is pretty reasonable and this will let us find out if we do in fact have to make that change. At a glance, it seems like a bit of a "it almost-never _really_ happens and there's no reasonable response" scenario, in which case failing fast is better?